wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/Command.java

```diff
@@ -572,11 +573,19 @@ public boolean runsWhenDisabled() {
   /**
    * Decorates this Command with a name.
    *
+   * <p>The returned commands `Execute` method is implicitly timed using the {@link Tracer} class.
+   *
    * @param name name
    * @return the decorated Command
    */
   public WrapperCommand withName(String name) {
-    WrapperCommand wrapper = new WrapperCommand(Command.this) {};
+    WrapperCommand wrapper =
+        new WrapperCommand(Command.this) {
+          @Override
+          public void execute() {
+            Tracer.traceFunc(name + "Execute", super::execute);
+          }
+        };
```

Comment:
Why is this added?

Reply:
Tracing commands returned from factory methods isn't very helpful, since it'll be under names like `SequentialCommandGroup` and there will probably be multiple of them, so tracing commands is opt-in by default. Tracing commands can be opted into by naming the command.

Reply:
I don't like that behavior- it feels arbitrary and random (some library commands add names). I would add a factory or decorator that adds tracing to a command instead.

Reply:
It was discussed a bit in [discord](https://discord.com/channels/176186766946992128/528555967827148801/1288356785140404254), the issue with the tracing is the command really should be named when doing so. This would likely be expressed by a decorator like `.traced(string name)`.

Reply:
Added the new command decorator and reverted the functionality fo `withName(String)`